extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/datatransferproject/datatransfer/flickr/media/FlickrMediaImporter.java

```diff
@@ -0,0 +1,332 @@
+                "Error: There is no data to import");
+
+        if (data.getAlbums() != null) {
+            storeAlbums(jobId, data.getAlbums());
+        }
+
+        if (data.getPhotos() != null) {
+            for (PhotoModel photo : data.getPhotos()) {
+                try {
+                    importSinglePhoto(idempotentExecutor, jobId, photo);
+                } catch (FlickrException e) {
+                    if (e.getMessage().contains("Upload limit reached")) {
+                        throw new DestinationMemoryFullException("Flickr destination memory reached", e);
+                    } else if (e.getMessage().contains("Photo already in set")) {
+                        // This can happen if we got a server error on our end, but the request went through.
+                        // When our retry strategy kicked in the request was complete and the photo already
+                        // uploaded
+                        continue;
+                    }
+                    throw new IOException(e);
+                }
+            }
+        }
+
+        if (data.getVideos() != null) {
+            for (VideoModel video : data.getVideos()) {
+                try {
+                    importSingleVideo(idempotentExecutor, jobId, video);
+                } catch (FlickrException e) {
+                    if (e.getMessage().contains("Upload limit reached")) {
+                        throw new DestinationMemoryFullException("Flickr destination memory reached", e);
+                    } else if (e.getMessage().contains("Photo already in set")) {
+                        // This can happen if we got a server error on our end, but the request went through.
+                        // When our retry strategy kicked in the request was complete and the photo already
+                        // uploaded
+                        continue;
+                    }
+                    throw new IOException(e);
+                }
+            }
+        }
+
+        return new ImportResult(ImportResult.ResultType.OK);
+    }
+
+    // Store any album data in the cache because Flickr only allows you to create an album with a
+    // photo in it, so we have to wait for the first photo to create the album
+    private void storeAlbums(UUID jobId, Collection<MediaAlbum> albums) throws IOException {
+        for (MediaAlbum album : albums) {
+            jobStore.create(
+                    jobId,
+                    ORIGINAL_ALBUM_PREFIX + album.getId(),
+                    new FlickrTempPhotoData(album.getName(), album.getDescription()));
+        }
+    }
+
+    private void importSinglePhoto(
+            IdempotentImportExecutor idempotentExecutor, UUID id, PhotoModel photo) throws Exception {
+        String photoId =
+                idempotentExecutor.executeAndSwallowIOExceptions(
+                        photo.getAlbumId() + "-" + photo.getDataId(),
```

Comment:
```suggestion
                        photo.getIdempotentId(),
```

here and on line 199 below (for video) might be better to prefer the model's own logic for this - it *should* already be available to you as [`PhotoModel#getIdempotentId`](https://sourcegraph.com/github.com/google/data-transfer-project@4eaba12604aa1ab6d4cf691e3eca5206948f7bb8/-/blob/portability-types-common/src/main/java/org/datatransferproject/types/common/models/photos/PhotoModel.java?L186:17-186:32) and `VideoModel#getIdempotentId` (via [`MediaObject` via `CreativeWork` via `Thing#getIdempotentId` (yayy java)](https://sourcegraph.com/github.com/google/data-transfer-project@4eaba12604aa1ab6d4cf691e3eca5206948f7bb8/-/blob/portability-types-common/src/main/java/org/datatransferproject/types/common/models/Thing.java?L42&subtree=true))

[example usage line here](https://sourcegraph.com/github.com/google/data-transfer-project@4eaba12/-/blob/extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/datatransferproject/transfer/microsoft/media/MicrosoftMediaImporter.java?L195&subtree=true) (though it's [indirect, via `DownloadableFile`](https://sourcegraph.com/github.com/google/data-transfer-project@4eaba12604aa1ab6d4cf691e3eca5206948f7bb8/-/blob/extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/datatransferproject/transfer/microsoft/media/MicrosoftMediaImporter.java?L109-114&subtree=true#tab=def))

Reply:
Changed that, thanks